### PR TITLE
adding new concepts to valueset

### PR DESCRIPTION
**Pull request by ChavanShweta**
OP-1126
OP-1100
Reference #508
@burrowse  Please review. Thanks.

### CODE REVIEW

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 1]
```diff
@@ -6183,4 +6182,22 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
 tobacco type,01,,4219336,Cigarette smoking tobacco,4282779
 tobacco type,01,,4219336,Pipe smoking tobacco,4224317
 tobacco type,01,,4219336,Smoked tobacco only,4298794
-Encounter type,OA,,581399,Telehealth,
+vx_manufacturer_source,SOL,Solvay Pharmaceuticals,,,
+Encounter type,TH,,581399,Telehealth,
+Facility type,HOSPITAL_OUTPATIENT_HEMONC_CLINIC,HOSPITAL_OUTPATIENT_CLINIC-AMBULATORY_CARE-subset,8756,outpatient hospital Hematology Oncology,45756811
+Facility type,HOSPITAL_OUTPATIENT_CARDIOLOGY_CLINIC,HOSPITAL_OUTPATIENT_CLINIC-AMBULATORY_CARE-subset,8756,Outpatient Hospital Cardiology,45756805
+Facility type,HOSPITAL_OUTPATIENT_NEUROSURGERY_CLINIC,HOSPITAL_OUTPATIENT_CLINIC-AMBULATORY_CARE-subset,8756,Outpatient Hospital Neurosurgery,38004459
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6189]: HOSPITAL_OUTPATIENT_OBGYN_CLINIC was also added to the facility type valueset for v6 as well.

Add the following:

Facility type,HOSPITAL_OUTPATIENT_OBGYN_CLINIC,HOSPITAL_OUTPATIENT_CLINIC-AMBULATORY_CARE-subset,8756,Obstetrics and Gynecology,38004461

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 2]
```diff
@@ -6183,4 +6182,22 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
 tobacco type,01,,4219336,Cigarette smoking tobacco,4282779
 tobacco type,01,,4219336,Pipe smoking tobacco,4224317
 tobacco type,01,,4219336,Smoked tobacco only,4298794
-Encounter type,OA,,581399,Telehealth,
+vx_manufacturer_source,SOL,Solvay Pharmaceuticals,,,
+Encounter type,TH,,581399,Telehealth,
+Facility type,HOSPITAL_OUTPATIENT_HEMONC_CLINIC,HOSPITAL_OUTPATIENT_CLINIC-AMBULATORY_CARE-subset,8756,outpatient hospital Hematology Oncology,45756811
+Facility type,HOSPITAL_OUTPATIENT_CARDIOLOGY_CLINIC,HOSPITAL_OUTPATIENT_CLINIC-AMBULATORY_CARE-subset,8756,Outpatient Hospital Cardiology,45756805
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6188]: Add concept=38004451 mapping for cardiology as well.

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 3]
```diff
@@ -6183,4 +6182,22 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
 tobacco type,01,,4219336,Cigarette smoking tobacco,4282779
 tobacco type,01,,4219336,Pipe smoking tobacco,4224317
 tobacco type,01,,4219336,Smoked tobacco only,4298794
-Encounter type,OA,,581399,Telehealth,
+vx_manufacturer_source,SOL,Solvay Pharmaceuticals,,,
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6185]: Additional manufacturers need to be added:

vx_manufacturer_source,BN,Bavarian Nordic A/S,,,
vx_manufacturer_source,TVA,TEVA Pharmaceuticals USA,,,

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 4]
```diff
@@ -6183,4 +6182,22 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
...
+result_qual_source,AB NEG,AB Neg,,AB Negative,
+result_qual_source,AB NOT DETECTED,Ab not detected,,AB Not detected,
+result_qual_source,AB POS,AB Pos,,AB POSITIVE,
+result_qual_source,B NEG,B Neg,,B Neg,
+result_qual_source,B POS,B Pos,,B Pos,
+result_qual_source,INCONCLUSIVE,Inconclusive,,Inconclusive,
+result_qual_source,INFLUENZA A VIRUS,Influenza A virus,,Influenza A virus,
+result_qual_source,INFLUENZA B VIRUS,Influenza B virus,,Influenza B virus,
+result_qual_source,O NEG,O Neg,,O NEGATIVE,
+result_qual_source,B POS,B Pos,,O positive,
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6201]: Looks like a typo, is this supposed to say:

result_qual_source,O POS,O Pos,,O positive,

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 5]
```diff
@@ -6183,4 +6182,22 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
...
+result_qual_source,SARS CORONAVIRUS 2,SARS coronavirus 2,,SARS coronavirus,
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6202]: This should be updated to say SARS coronavirus 2 throughout we wouldn't want to mis-map 2 to 1

result_qual_source,SARS CORONAVIRUS 2,SARS coronavirus 2,,SARS coronavirus 2,

**Review by burrowse — CHANGES_REQUESTED** [review]
Hey @ChavanShweta Thanks for working on this! See comments for the valuesets included in this PR.

**Comment by burrowse** [comment]
@ChavanShweta Please add these provider specialty maps... most codes are not in the vocabulary as yet but we should add them proactively since they are included in PCORnet's valueset.
[provider_specialty_maps.xlsx](https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/files/5556228/provider_specialty_maps.xlsx)

**Comment by burrowse** [comment]
@ChavanShweta Please add the following line for the language update to be proactive as well:

source_concept_class,target_concept, pcornet_name,source_concept_id,concept_description
Language,CNR,Montenegrin,,,

**Comment by burrowse** [comment]
Work in progress for result unit mapping. Will add result unit source mapping asap.
[units_valueset_update.xlsx](https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/files/5556487/units_valueset_update.xlsx)

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv (hunk 2 above)
Comment (ChavanShweta) [reply v4.0_to_v6.0/sql_etl/data/concept_map.csv:6188]: already present in mapping.

Comment (burrowse) [reply v4.0_to_v6.0/sql_etl/data/concept_map.csv:6188]: Can you point to a line? I didn't see it captured.

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 6]
```diff
@@ -6183,4 +6182,83 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
...
+Provider Specialty,133VN1401X,Dietitian Registered Nutrition Oncology,,Dietitian Registered Nutrition Oncology,
+Provider Specialty,133VN1401X,Dietitian Registered Nutrition Pediatric Critical Care,,Dietitian Registered Nutrition Pediatric Critical Care,
+Provider Specialty,133VN1501X,Dietitian Registered Nutrition Sports Dietetics,,Dietitian Registered Nutrition Sports Dietetics,
+Provider Specialty,207RA0002X,Internal Medicine Adult Congenital Heart Disease,,Internal Medicine Adult Congenital Heart Disease,
+Provider Specialty,207WX0120X,Ophthalmology Cornea and External Diseases Speci,903243,Ophthalmology Cornea and External Diseases Speci,
+Provider Specialty,2083A0300X,Preventive Medicine Addiction Medicine,,Preventive Medicine Addiction Medicine,
+vx_manufacturer_source,BN,Bavarian Nordic A/S,,,
+vx_manufacturer_source,TVA,TEVA Pharmaceuticals USA,,,
+Facility type,HOSPITAL_OUTPATIENT_OBGYN_CLINIC,HOSPITAL_OUTPATIENT_CLINIC-AMBULATORY_CARE-subset,8756,Obstetrics and Gynecology,38004461
+result_qual_source,O POS,B Pos,,O positive,
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6264]: Still looks like this says B Pos in the third column...

**Comment by burrowse** [comment]
@ChavanShweta Please see attached a result_unit_source valueset update.
[result_unit_source.xlsx](https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/files/5563456/result_unit_source.xlsx)

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 7]
```diff
@@ -6183,4 +6182,83 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
...
+Route,SURGICAL_DRAIN,Surgical  drain  route  (qualifier  value),4304412,abdominal drain,
+Route,TRANSMUCOSAL,Transmucosal  route  (qualifier  value),4232601,Transmucosal,
+Route,ILEOSTOMY,Ileostomy  route  (qualifier  value),4305679,retrograde ileostomy,
+Route,NASOJEJUNAL,Nasojejunal  route  (qualifier  value),4305834,Nasojejunal Tube,
+Route,EXTRACORPOREAL,Extracorporeal  route  (qualifier  value),37018288,extracorporeal,
+Route,INTRAVASCULAR,Intravascular  route  (qualifier  value),40492287,Intravascular,
+Route,INTRAPERICARDIAL,Intrapericardial  route  (qualifier  value),40492305,Intrapericardial,
+Route,INTRAHEPATIC,Intrahepatic  route  (qualifier  value),40493258,intrahepatic,
+Route,EXTRACORPOREAL_HEMODIALYSIS,Extracorporeal hemodialysis route (qualifier value),44801748,Haemodiafiltration route,
+Route,ENDOCERVICAL,Endocervical  route  (qualifier  value),45956872,cervical,
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6234]: Add Route,EPILESIONAL,Epilesional route (qualifier value),35608078,Epilesional route
Add Route,EPILESIONAL,Epilesional route (qualifier value),45956880,Epilesional
Add  Route, INFILTRATION,Infiltration route (qualifier value),37397638,Infiltration route
Add  Route,INTESTINAL,Intestinal route (qualifier value),37207459,Intestinal route
Add  Route,INTRACERVICAL,Intracervical route (qualifier value),45956872,Intracervical
Add  Route,INTRACHOLANGIOPANCREATIC,Intracholangiopancreatic route (qualifier value),37207461,Intracholangiopancreatic route
Add  Route,INTRAPORTAL,Intraportal route (qualifier value),37207462,Intraportal route
Add  Route,OCULAR,Ocular route (qualifier value),4184451,Ocular
Add  Route,PERITUMORAL,Peritumoral route (qualifier value),37207463,Peritumoral route
Add  Route,POSTERIOR_JUXTASCLERAL,Posterior juxtascleral route (qualifier value),37207464,Posterior juxtascleral route
Add  Route,SUBRETINAL,Subretinal route (qualifier value),36703536,Subretinal route
Add  Route,SUBRETINAL,Subretinal route (qualifier value),37207465,Subretinal

**Review by burrowse — CHANGES_REQUESTED** [review]

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 8]
```diff
@@ -6183,4 +6182,129 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
...
+Rx Dose Form,VAGINAL_SYSTEM,VAGINAL_SYSTEM=Vaginal System,37496609,Vaginal System,
+Rx Dose Form,ORAL_FILM,ORAL_FILM=Oral Film,37498347,Oral Film,
+Rx Dose Form,SUBLINGUAL_FILM,SUBLINGUAL_FILM=Sublingual Film,37498348,Sublingual Film,
+Rx Dose Form,SUBLINGUAL_POWDER,SUBLINGUAL_POWDER=Sublingual Powder,40220762,Sublingual Powder,
+Rx Dose Form,MEDICATED_PATCH,MEDICATED_PATCH=Medicated Patch,42629089,Medicated Patch,
+Rx Dose Form,INJECTABLE_FOAM,INJECTABLE_FOAM=Injectable Foam,44784844,Injectable Foam,
+Rx Dose Form,POWDER_FOR_NASAL_SOLUTION,POWDER_FOR_NASAL_SOLUTION=Powder for Nasal Solution,45775488,Powder for Nasal Solution,
+Rx Dose Form,POWDER_FOR_ORAL_SOLUTION,POWDER_FOR_ORAL_SOLUTION=Powder for Oral Solution,45775491,Powder for Oral Solution,
+Rx Dose Form,POWDER_FOR_ORAL_SUSPENSION,POWDER_FOR_ORAL_SUSPENSION=Powder for Oral Suspension,45775492,Powder for Oral Suspension,
+Rx Dose Form,TOPICAL_SUSPENSION,TOPICAL_SUSPENSION=Topical Suspension,46234410,Topical Suspension,
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6302]: Missing the following please add:

Rx Dose Form,POWDER_FOR_PYELOCALYCEAL_SOLUTION,POWDER_FOR_PYELOCALYCEAL_SOLUTION=Powder for Pyelocalyceal Solution,1146250,Powder for Pyelocalyceal Solution,
Rx Dose Form,MUCOUS_MEMBRANE_TOPICAL_SOLUTION,Mucous  Membrane  Topical  Solution,19082110,Mucous Membrane Topical Solution,
Rx Dose Form,CHEWABLE_EXTENDED_RELEASE_TABLET,Chewable Extended Release Tablet,37498346,Chewable Extended Release Oral Tablet,
Rx Dose Form,DELAYED_RELEASE_CAPSULE,Delayed Release Capsule,19082255,Delayed Release Oral Capsule,
Rx Dose Form,DELAYED_RELEASE_GRANULES,Delayed Release Granules,37499224,Delayed Release Oral Granules,
Rx Dose Form,DELAYED_RELEASE_TABLET,Delayed Release Tablet,19001949,Delayed Release Oral Tablet,
Rx Dose Form,EFFERVESCENT_TABLET,Effervescent Tablet,44817840,Effervescent Oral Tablet,
Rx Dose Form,GRANULES_FOR_ORAL_SOLUTION,Granules for Oral Solution,45775489,Granules for Oral Solution,
Rx Dose Form,GRANULES_FOR_ORAL_SUSPENSION,Granules for Oral Suspension,45775490,Granules for Oral Suspension,
Rx Dose Form,INHALANT_SPRAY,Inhalant Spray,37499223,Inhalation Spray,
Rx Dose Form,INHALANT_SUSPENSION,Inhalant Suspension,1356078,Inhalation Suspension,

**Review by burrowse — CHANGES_REQUESTED** [review]

**Review by burrowse — CHANGES_REQUESTED** [review]
See comments for changes needed

File: v4.0_to_v6.0/sql_etl/scripts/etl_scripts/h_condition.sql [hunk 9]
```diff
@@ -102,7 +102,7 @@ cast(visit_occurrence_id as text) as encounterid,
 observation_date as report_date,
 cast (null as date) as resolve_date,
 cast (null as date) as onset_date,
-'CC'  as condition_status,
+'AC'  as condition_status,
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/scripts/etl_scripts/h_condition.sql:105]: AC is good here for condition_status.

Chief Compliant records should have 'CC' as condition_source though (line 108).

**Review by burrowse — CHANGES_REQUESTED** [review]

**Comment by ChavanShweta** [comment]
@burrowse 
https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/pull/510#discussion_r532679394
I have made the changes in the [condition_source](https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/pull/510/commits/6a40489ba0106655867ecc162b58d334401ea681) for the chief complaints but the recent ETL documentation for pcornet doesn't say this valueset exists unless this is specific to COVID data.
[
Screen Shot 2020-11-30 at 10 29 24 AM
](url)

**Comment by burrowse** [comment]
> @burrowse
> [#510 (comment)](https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/pull/510#discussion_r532679394)
> I have made the changes in the [condition_source](https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/pull/510/commits/6a40489ba0106655867ecc162b58d334401ea681) for the chief complaints
> 
> https://github.com/PEDSnet/pedsnetcdm_to_pcornetcdm/blob/2293ca56e25b2c6578a660d640d80086b4fb7482/v4.0_to_v6.0/sql_etl/scripts/etl_scripts/h_condition.sql#L108
> 
> but the recent ETL documentation for pcornet doesn't say this valueset exists unless this is specific to COVID data.
> [ Screen Shot 2020-11-30 at 10 29 24 AM](url)

Thanks @ChavanShweta ! What version are you referencing? Looking at the v6 10/22 PDF CC is there. Did they make an update after?

Screen Shot 2020-11-30 at 10 41 05 AM

File: v4.0_to_v6.0/sql_etl/data/concept_map.csv [hunk 10]
```diff
@@ -6183,4 +6149,224 @@ tobacco type,01,,4219336,Cigar smoking tobacco,4132133
...
+Rx Dose Form,DELAYED_RELEASE_CAPSULE,Delayed Release Capsule,19082255,Delayed Release Oral Capsule,
+imm_body_site_source,RA ,RA=right arm,R ARM,,
+Rx Dose Form,DELAYED_RELEASE_GRANULES,Delayed Release Granules,37499224,Delayed Release Oral Granules,
+Rx Dose Form,DELAYED_RELEASE_TABLET,Delayed Release Tablet,19001949,Delayed Release Oral Tablet,
+Rx Dose Form,EFFERVESCENT_TABLET,Effervescent Tablet,44817840,Effervescent Oral Tablet,
+Rx Dose Form,GRANULES_FOR_ORAL_SOLUTION,Granules for Oral Solution,45775489,Granules for Oral Solution,
+Rx Dose Form,GRANULES_FOR_ORAL_SUSPENSION,Granules for Oral Suspension,45775490,Granules for Oral Suspension,
+Rx Dose Form,INHALANT_SPRAY,Inhalant Spray,37499223,Inhalation Spray,
+Rx Dose Form,INHALANT_SUSPENSION,Inhalant Suspension,1356078,Inhalation Suspension,
+Rx Dose Form,MUCOUS_MEMBRANE_TOPICAL_SOLUTION,Mucus Membrane Topical Solution,19082110,Mucous Membrane Topical Solution,null
```
Comment (burrowse) [inline v4.0_to_v6.0/sql_etl/data/concept_map.csv:6290]: Update to say MUCUOUS_MEMBRANE_TOPICAL_SOLUTION

**Review by burrowse — CHANGES_REQUESTED** [review]

**Review by burrowse — APPROVED** [review]

**Pull request merged**